pkgs/development/haskell-modules/configuration-ghc-9.0.x.nix

```
@@ -1,12 +1,13 @@
-{ }:
-
+{ haskellLib }:
+with haskellLib;
 self: super: {
   # Disable GHC 9.0.x core libraries.
   array = null;
   base = null;
   binary = null;
   bytestring = null;
   Cabal = null;
+  Cabal-syntax = null;
```

You'll need to reintroduce the polyfill package for this via `configuration-hackage2nix/main.yaml`.

```suggestion
  Cabal-syntax = self.Cabal-syntax_3_6_0_0;
```